File: solutions/3924-count-odd-letters-from-number/count-odd-letters-from-number.py
class Solution:
    def countOddLetters(self, n: int) -> int:

        ord = [16577, 97, 4672, 648, 1218, 2067, 8464, 2336, 541, 17]
        toggle = 0

        while n:
            n, digit = divmod(n, 10)
            toggle^= ord[digit]

        return toggle.bit_count()    

File: solutions/3924-count-odd-letters-from-number/count-odd-letters-from-number.cpp
class Solution {
public:
    int countOddLetters(int n) {
        int d[10] = {16577, 97, 4672, 648, 1218, 2067, 8464, 2336, 541, 17};
        int toggle = 0;

        while (n > 0) {
            int digit = n % 10;
            toggle ^= d[digit];
            n /= 10; }

        return __builtin_popcount(toggle); }
};